packages/devtools_app/test/flutter/service_extension_widgets_test.dart

```diff
@@ -110,6 +111,59 @@ void main() {
       expect(restarts, 0);
     });
   });
+
+  group('Structured Errors checkbox', () {
+    StreamSubscription serviceState;
+    ServiceExtensionState mostRecentState;
+    setUp(() {
+      mockServiceManager.serviceExtensionManager.fakeFrame();
+      serviceState =
+          mockServiceManager.serviceExtensionManager.getServiceExtensionState(
+        structuredErrors.extension,
+        (data) {
+          mostRecentState = data;
+        },
+      );
+    });
+
+    tearDown(() async {
+      await serviceState.cancel();
+    });
+    testWidgets('toggles', (WidgetTester tester) async {
+      await mockServiceManager.serviceExtensionManager
+          .fakeAddServiceExtension(structuredErrors.extension);
+
+      final button = StructuredErrorsCheckbox();
+      await tester.pumpWidget(wrap(Scaffold(body: Center(child: button))));
+      expect(find.byWidget(button), findsOneWidget);
+      await tester.tap(find.byWidget(button));
+      await tester.pumpAndSettle();
+      mockServiceManager.serviceExtensionManager.fakeFrame();
+      expect(mostRecentState.value, true);
+      await tester.tap(find.byWidget(button));
+      await tester.pumpAndSettle();
+      expect(mostRecentState.value, false);
+    });
+
+    testWidgets('updates based on the service extension',
+        (WidgetTester tester) async {
+      await mockServiceManager.serviceExtensionManager
+          .fakeAddServiceExtension(structuredErrors.extension);
+      final button = StructuredErrorsCheckbox();
+      await tester.pumpWidget(wrap(Scaffold(body: Center(child: button))));
+      expect(find.byWidget(button), findsOneWidget);
+
+      await mockServiceManager.serviceExtensionManager
+          .setServiceExtensionState(structuredErrors.extension, true, true);
+      await tester.pumpAndSettle();
+      expect(checkbox.value, true, reason: 'The extension is enabled.');
+
+      await mockServiceManager.serviceExtensionManager
+          .setServiceExtensionState(structuredErrors.extension, false, false);
+      await tester.pumpAndSettle();
+      expect(checkbox.value, false, reason: 'The extension is disabled.');
```

Comment:
@kenzieschmoll @albertusangga @terrylucas 
using `reason` like this is a great way to clarify what expect statements in tests are trying to test.